packages/web-components/fast-foundation/src/select/select.ts

```diff
@@ -0,0 +1,374 @@
+        return selectedOption || null
+    }
+
+    /**
+     * When a user clicks on an option we need to update the known
+     * parts of
+     * - currently selected value
+     * - array of values
+     */
+    /* eslint-disable-next-line @typescript-eslint/no-unused-vars */
+    public optionSelectionChange(value: string): void {
+        this.handleMultiple(value);
+
+        this.options.forEach(element  => {
+            if (element.value != value) {
+                const el: any = element;
+                el.removeAttribute('current');
+                this.activeOptionIndex = Array.from(this.options).indexOf(element);
+            }
+        });
+
+        this.updateSelectValue(value);
+        this.typeAheadValue = '';
+        this.open = !this.open;
+        console.log("Call six")
+        // Focus back on the select when menu closes
+        this.button.focus()
+    }
+
+    /**
+     * When the author leverages the slot we need to ensure that the a11y and
+     * functionality that is tied to the given part still function as designed
+     */
+    public registerButtonSlotChange(): void {
+        let slot = this.shadowRoot!.querySelector('slot[name=button-container]');
+        if (slot) {
+            slot.addEventListener('slotchange', () => {
+                this.applyButtonControllerCode();
+            });
+        }
+    }
+
+    private applyButtonControllerCode(): void {
+        if (this.button) {
+            console.log("This hits", this.button)
+            this.button.setAttribute('tabindex', '0');
+            this.button.setAttribute('aria-haspopup', 'listbox');
+            this.button.setAttribute('aria-expanded', this.open ? 'true' : 'false');
+            this.button.setAttribute('role', 'button');
+
+            this.button.addEventListener('click', this.clickHandler);
+            this.button.addEventListener('keydown', this.keypressHandlerButton);
+        }
+    }
+
+    private applyListboxControllerCode(): void {
+        this.listbox.forEach((listbox: HTMLElement, index: number) => {
+            listbox.setAttribute("role", "listbox");
+            listbox.addEventListener("keydown", this.keypressHandlerListbox);
+            listbox.addEventListener("focusout", this.focusoutHandlerListbox);
+        })
+    }
+
+    private regexEscape(str) {
+        return str.replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&')
+    }
+
+    /**
+     * Move focus to an option whose label matches characters typed by the user.
+     * Consecutive keystrokes are batched into a buffer of search text used
+     * to match against the set of options.  If TYPE_AHEAD_TIMEOUT_MS passes
+     * between consecutive keystrokes, the search restarts.
+     * 
+     * @param typedKey
+     */
+    private typeAheadValue: string = '';
+    private typeAheadTimeoutHandler: number = -1;
+    private typeAheadExpired: boolean = false;
+    private static readonly TYPE_AHEAD_TIMEOUT_MS = 1000;
+    public handleTypeAhead(typedKey) {
```

Comment:
Do we need to have type ahead as part of the initial feature set?  Seems like an extra.  There should be an option to enable/disable it at least.

Reply:
I think we should look to include basic type ahead here - I don't think we need full combobox behavior. 

I'm definitely in favor of this being configurable per @scomea's suggestion. I'd suggest including a *property* for the timeout itself so that's configurable as well. I know most browsers tend to use `500ms` for this from some Open UI research we just discussed over there.